app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt

```diff
@@ -0,0 +1,181 @@
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) {
+      return
+    }
+    hideKeyboard()
+
+    TheActivity.component.inject(this)
+
+    val screenKey = screenRouter.key<ShortCodeSearchResultScreenKey>(this)
+    setupToolBar(screenKey)
+    setupScreen()
+
+    val uiStateProducer = ShortCodeSearchResultStateProducer(
+        shortCode = screenKey.shortCode,
+        patientRepository = patientRepository,
+        userSession = userSession,
+        facilityRepository = facilityRepository,
+        bloodPressureDao = bloodPressureDao,
+        ui = this,
+        schedulersProvider = schedulersProvider
+    )
+    binding = ViewControllerBinding.bindToView(this, uiStateProducer, uiChangeProducer)
+    setupClickEvents()
+  }
+
+  override fun onDetachedFromWindow() {
+    if (::disposable.isInitialized) disposable.dispose()
+    super.onDetachedFromWindow()
+  }
+
+  private fun setupToolBar(screenKey: ShortCodeSearchResultScreenKey) {
+    val shortCode = screenKey.shortCode
+
+    // This is guaranteed to be exactly 7 characters in length.
+    val prefix = shortCode.substring(0, 3)
+    val suffix = shortCode.substring(3)
+
+    val formattedShortCode = "$prefix${Unicode.nonBreakingSpace}$suffix"
+
+    val textSpacingSpan = TextAppearanceWithLetterSpacingSpan(context, R.style.Clinic_V2_TextAppearance_Body0Left_NumericBold_White100)
+
+    toolBar.title = Truss()
+        .pushSpan(textSpacingSpan)
+        .append(formattedShortCode)
+        .popSpan()
+        .build()
+
+    toolBar.setNavigationOnClickListener {
+      screenRouter.pop()
+    }
+    toolBar.setOnClickListener {
+      screenRouter.pop()
+    }
+  }
+
+  private fun setupClickEvents() {
+    newPatientButton.setOnClickListener { binding.onEvent(SearchPatient) }
+    disposable = downstreamEvents
+        .ofType<SearchResultClicked>()
+        .map { binding.onEvent(ViewPatient(it.patientUuid)) }
+        .subscribe()
+  }
+
+  private fun setupScreen() {
+    newPatientButton.text = resources.getString(R.string.shortcodesearchresult_enter_patient_name_button)
+
+    resultsRecyclerView.layoutManager = LinearLayoutManager(context)
+    resultsRecyclerView.adapter = adapter
+  }
+
+  override fun openPatientSummary(patientUuid: UUID) {
+    screenRouter.push(PatientSummaryScreenKey(
+        patientUuid = patientUuid,
+        intention = OpenIntention.ViewExistingPatient,
+        screenCreatedTimestamp = Instant.now(utcClock)
+    ))
+  }
+
+  override fun openPatientSearch() {
+    screenRouter.push(PatientSearchScreenKey())
+  }
+
+  override fun showLoading() {
+    loader.visibility = View.VISIBLE
+  }
+
+  override fun hideLoading() {
+    loader.visibility = View.GONE
+  }
+
+  override fun showSearchResults(foundPatients: PatientSearchResults) {
+    val items = SearchResultsItemType
+        .from(foundPatients)
+    items.map {
+      it.uiEvents = downstreamEvents
+    }
```

Comment:
We could use this to achieve what you are trying to achieve.

```kotlin
val items = SearchResultsItemType
        .from(foundPatients)
        .onEach { it.uiEvents = downstreamEvents }
```

Reply:
Yes. This is what I was looking for.